crates/transaction-pool/src/validate/eth.rs

```diff
@@ -309,12 +309,31 @@ where
                 )
             }
 
+            if transaction.to().is_none() {
+                return TransactionValidationOutcome::Invalid(
+                    transaction,
+                    Eip7702PoolTransactionError::MissingEip7702To.into(),
+                )
+            }
```

Comment:
this we don't really need because we assome the to field is mandatory, which is enforced for the `TxEip7702` tx.

so we can omit this